### PR TITLE
Add Collatz Conjecture exercise

### PR DESCRIPTION
**Pull request by RealBarrettBrown**
Okay, so it's been a year and I need help with `config.json`.  What's this `uuid`?

### CODE REVIEW

**Comment by patricksjackson** [comment]
I'm not sure exactly what the `uuid` is used for, but it needs to be a unique identifier for the exercise. It looks like you already used the `configlet` tool to generate the README (?) you can also use it to generate a `uuid` to add in as well.

If that makes no sense somebody can explain further. I don't think `configlet` is documented very much, if at all, so it's not exactly obvious what to do.

File: exercises/collatz-conjecture/test/test_collatz_conjecture.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
```
Comment (patricksjackson) [inline exercises/collatz-conjecture/test/test_collatz_conjecture.c]: Do you need all of these include files?

File: exercises/collatz-conjecture/src/example.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   int number = start;
+
+   if (start <= 0) {
+      return ERROR_VALUE;
+   }
+
+   while (number != 1) {
+      step++;
+
+      if (is_even(number)) {
```
Comment (patricksjackson) [inline exercises/collatz-conjecture/src/example.c]: As a nitpick on the solution - it seems unnecessary to have a separate function for testing if a number is even.

**Comment by ryanplusplus** [comment]
@RealBarrettBrown I've been using this: https://www.uuidgenerator.net/

File: exercises/collatz-conjecture/src/example.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#include <stdbool.h>
+
+static bool is_even(int number)
+{
+   if (number % 2) {
+      return false;
+   }
+
+   return true;
+}
```
Comment (ryanplusplus) [inline exercises/collatz-conjecture/src/example.c]: `return `number % 2` == 0` (if this function even survives)

File: exercises/collatz-conjecture/test/test_collatz_conjecture.c (hunk 1 above)
Comment (ryanplusplus) [reply exercises/collatz-conjecture/test/test_collatz_conjecture.c]: As far as I can tell, none are necessary

Comment (RealBarrettBrown) [reply exercises/collatz-conjecture/test/test_collatz_conjecture.c]: Nope, they're not necessary.  Copied from another exercise.

File: exercises/collatz-conjecture/src/example.c (hunk 2 above)
Comment (RealBarrettBrown) [reply exercises/collatz-conjecture/src/example.c]: I agree

**Comment by ryanplusplus** [comment]
LGTM 👍

**Comment by patricksjackson** [comment]
👍

**Comment by patricksjackson** [comment]
I knew I missed something - this doesn't have the TEST_IGNORE() macros on the tests.

**Pull request merged**